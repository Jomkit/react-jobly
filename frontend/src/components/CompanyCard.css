.CompanyCard-card:hover {
    color: blue;
}