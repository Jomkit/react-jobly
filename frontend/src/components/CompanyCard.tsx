import { Link } from 'react-router-dom';
import { Card, CardBody, CardText, CardTitle } from 'reactstrap';
import './CompanyCard.css';

interface companyPropsInterface {
    handle: string;
    title: string;
    description: string;
    logoUrl?: string;
}

const CompanyCard = ({handle, title, description, logoUrl}: companyPropsInterface) => {
  return (
    <li>
        <Link to={`/companies/${handle}`}
            className='text-reset 
                text-decoration-none'
        >
            <Card className='CompanyCard-card p-3'>
                <CardTitle tag="h3">
                    {title}
                </CardTitle>
                <CardBody>
                    <CardText>{description}</CardText>
                    {logoUrl ? 
                    <img
                        alt="company logo"
                        src={logoUrl} /> :
                    <i className="fa-solid fa-image-portrait" style={{ fontSize: '100px' }}></i>
                    }
                </CardBody>
            </Card>
        </Link>
    </li>
  )
}

export default CompanyCard