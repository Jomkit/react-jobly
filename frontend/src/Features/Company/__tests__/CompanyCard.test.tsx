import { render } from "@testing-library/react";
import CompanyCard from "../CompanyCard";
import { MemoryRouter } from "react-router-dom";
import { FAKEDATA } from "../../../__tests__/_testCommon";

it('should render without crashing', () => { 
    const { handle, name, description, logoUrl } = FAKEDATA[0];
    render(
        <MemoryRouter>
            <CompanyCard key={handle} handle={handle} title={name} description={description} logoUrl={logoUrl} />
        </MemoryRouter>
    );
})

it('should match snapshot', () => { 
    let { handle, name, description, logoUrl } = FAKEDATA[0];
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard key={handle} handle={handle} title={name} description={description} logoUrl={logoUrl} />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
})