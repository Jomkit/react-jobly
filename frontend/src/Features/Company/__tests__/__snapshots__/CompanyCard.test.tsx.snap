// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <li>
    <a
      class="text-reset 
                text-decoration-none"
      href="/companies/test1"
    >
      <div
        class="CompanyCard-card p-3 m-5 border border-3 border-black card"
      >
        <h3
          class="card-title"
        >
          test company 1
        </h3>
        <div
          class="card-body"
        >
          <p
            class="card-text"
          >
            this is a test
          </p>
          <img
            alt="company logo"
            src="test logo"
          />
        </div>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
